rundeckapp/grails-app/services/rundeck/services/ScheduledExecutionService.groovy

```diff
@@ -693,10 +693,27 @@ class ScheduledExecutionService implements ApplicationContextAware, Initializing
         if(project) {
             results = results.withProject(project)
         }
+        def executionList = results.list()
+
+        def adhocRescheduleResult = rescheduleAdHocJobs(executionList)
+
+        [jobs: succeededJobs, failedJobs: failedJobs, executions: adhocRescheduleResult.executions, failedExecutions: adhocRescheduleResult.failedExecutions]
+    }
+
+    /**
+     * Reschedule the provided ad-hoc scheduled executions. Invalid executions will be cleaned up.
+     * @param serverUUID
+     * @param project
+     * @return
+     */
+    def rescheduleAdHocJobs(List<Execution> executionList) {
```

Comment:
my only concern is that this method name is confusing: `adhoc` is used to refer to Commands/executions not associated with a job. i believe it has also been used by the "run job later" mechanism, as it is essentially the same process to schedule those executions, but this method should be renamed, perhaps to something like `rescheduleOnetimeJobExecutions` , so using "onetime" instead of "adhoc" for jobs executions scheduled at a single time